setuptools/tests/config/test_setupcfg.py

```diff
@@ -716,6 +716,23 @@ def test_extras_require(self, tmpdir):
             }
             assert dist.metadata.provides_extras == set(['pdf', 'rest'])
 
+    @pytest.mark.parametrize(
+        "config",
+        [
+            "[options.extras_require]\nfoo = bar;python_version<'3'",
+            "[options.extras_require]\nfoo = bar;os_name=='linux'"
+        ]
+    )
+    def test_extras_warns_about_potential_misconfig(self, config, tmpdir):
```

Comment:
Let's also add tests that assert it does not warn for other cases?

Reply:
Done, thanks.